src/api/api.ts

```diff
@@ -179,41 +178,29 @@ const installedRequest = async (
     let response = await fetch(`__salesforce_instance_url__/${trimStart(url, "")}`, options);
 
     if ([400, 401].includes(response.status)) {
+        const refreshRequestOptions: RequestInit = {
+            method: "POST",
+            body: `grant_type=refresh_token&client_id=__client_key__&client_secret=__client_secret__&refresh_token=__global_access_token.json("[refreshToken]")__`,
+            headers: {
+                "Content-Type": "application/x-www-form-urlencoded",
+            },
+        };
+        
+        const refreshRes = await fetch(`__salesforce_instance_url__/services/oauth2/token`, refreshRequestOptions);
+        const refreshData = await refreshRes.json();
+
+        await client.setState<string>("oauth/global/accesstoken", refreshData.access_token, {
+            backend: true,
+        });
 
-        // todo: we need a better way of merge saving JSON encoded settings as this try/re-auth process will ALWAYS cause an
-        //  unnecessary extra request after the first access token has expired.
         options.headers = {
             ...options.headers,
             "Authorization": `Bearer [[oauth/global/accesstoken]]`,
         };
 
         response = await fetch(`__salesforce_instance_url__/${trimStart(url, "")}`, options);
-
-        if ([400, 401].includes(response.status)) {
-            const refreshRequestOptions: RequestInit = {
-                method: "POST",
-                body: `grant_type=refresh_token&client_id=__client_key__&client_secret=__client_secret__&refresh_token=__global_access_token.json("[refreshToken]")__`,
-                headers: {
-                    "Content-Type": "application/x-www-form-urlencoded",
-                },
-            };
-
-            const refreshRes = await fetch(`__salesforce_instance_url__/services/oauth2/token`, refreshRequestOptions);
-            const refreshData = await refreshRes.json();
-
-            await client.setState<string>("oauth/global/accesstoken", refreshData.access_token, {
-                backend: true,
-            });
-
-            options.headers = {
-                ...options.headers,
-                "Authorization": `Bearer [[oauth/global/accesstoken]]`,
-            };
-
-            response = await fetch(`__salesforce_instance_url__/${trimStart(url, "")}`, options);
```

Comment:
@iTweeno how does the original request work after a refresh if it's removed?

Reply:
what do you mean original request? That token will always be expired after x amount of time, so in this case the only request that is going to fail is the first one, and everytime it expired, because whenever it returns a 401 or 403, we immediately give it a correct token.
This way there is no need to always be fetching with the first token.

Reply:
This is similar to what im doing in devops, and it works perfectly

Reply:
@iTweeno not sure what you mean, surely the original request will still need to happen?

Reply:
the original request is the API endpoint the system was attempting to hit before a 401 was encountered